src/util/OrmUtils.ts

```diff
@@ -415,6 +323,39 @@ export class OrmUtils {
         return undefined
     }
 
+    public static isCriteriaNullOrEmpty(criteria: unknown): boolean {
+        return (
+            criteria === undefined ||
+            criteria === null ||
+            criteria === "" ||
+            (Array.isArray(criteria) && criteria.length === 0) ||
+            (OrmUtils.isPlainObject(criteria) &&
+                Object.keys(criteria).length === 0)
+        )
+    }
+
+    public static isSinglePrimitiveCriteria(
+        criteria: unknown,
+    ): criteria is SinglePrimitiveCriteria {
+        return (
+            typeof criteria === "string" ||
+            typeof criteria === "number" ||
+            criteria instanceof Date
+        )
+    }
+
+    public static isPrimitiveCriteria(
+        criteria: unknown,
+    ): criteria is PrimitiveCriteria {
+        if (Array.isArray(criteria)) {
+            return criteria.every((value) =>
+                OrmUtils.isSinglePrimitiveCriteria(value),
```

Comment:
nit: IMO it's more clear to use `this` when calling methods on this class. There's a few more in these changes
```suggestion
                this.isSinglePrimitiveCriteria(value),
```